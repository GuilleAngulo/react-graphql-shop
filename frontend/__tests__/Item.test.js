import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
    id: 'ADBC123',
    title: 'A Cool Item',
    price: 5000,
    description: 'This item is really cool',
    image: 'cool.jpg',
    largeImage: 'largecool.jpg',
};

describe('<Item/>', () => {

    it('renders and matches the snapshot', () => {
        const wrapper = shallow(<ItemComponent item= {fakeItem} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    
    it('renders the image properly', () => {
        const wrapper = shallow(<ItemComponent item= {fakeItem} />);
        const img = wrapper.find('img');
        expect(img.props().src).toBe(fakeItem.image);
        expect(img.props().alt).toBe(fakeItem.title);
    });
    
    it('renders the priceTag and title', () => {
        const wrapper = shallow(<ItemComponent item= {fakeItem} />);
        const PriceTag = wrapper.find('PriceTag');
        //console.log(PriceTag.text());
        //console.log(PriceTag.debug());
        //console.log(PriceTag.dive().text());
        //console.log(PriceTag.children());
        expect(PriceTag.children().text()).toBe('$50');
        expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
    });

    it('render out the buttons properly', () => {
        const wrapper = shallow(<ItemComponent item= {fakeItem} />);
        const buttonList = wrapper.find('.buttonList');
        expect(buttonList.children()).toHaveLength(3);
        expect(buttonList.find('Link')).toHaveLength(1);
        expect(buttonList.find('Link').exists()).toBe(true);
        expect(buttonList.find('Link').exists()).toBeTruthy();
        expect(buttonList.find('AddToCart').exists()).toBe(true);
        expect(buttonList.find('DeleteItem').exists()).toBe(true);
    }); 


});