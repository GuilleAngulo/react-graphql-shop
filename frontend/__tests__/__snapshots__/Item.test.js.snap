// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="cool.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ADBC123",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This item is really cool
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ADBC123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      item={
        Object {
          "description": "This item is really cool",
          "id": "ADBC123",
          "image": "cool.jpg",
          "largeImage": "largecool.jpg",
          "price": 5000,
          "title": "A Cool Item",
        }
      }
    />
    <DeleteItem
      id="ADBC123"
    >
      Delete this Item 🗑
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
